invokeai/app/invocations/baseinvocation.py

```diff
@@ -36,7 +36,7 @@ def get_all_subclasses_tuple(cls):
         return tuple(subclasses)
 
 
-class BaseInvocation(ABC, BaseModel):
+class BaseInvocation(ABC, InvokeAISettings):
```

Comment:
I am extremely hesitant to use BaseSettings as a base class for nodes, mostly because nodes aren't really "settings" of the application, and I'm worried about what complications we might encounter in Pydantic by trying to use the settings classes for this. I already had the "defaults" command in the CLI for this, though it's fairly rudimentary.

I do see the benefit to being able to specify defaults (I see what you put in config.py), but not sure if this is the right way to do it?

Reply:
Can you change settings at runtime? Does it allow you to change field defaults and such? So long as it doesn't allow you to create invalid nodes (which would make shared graphs not work), it might be okay.

Unfortunately, there's not a ton of automated tests built around the CLI or API yet. I don't know if it serializes differently, parses differently, or behaves differently in certain situations.

Reply:
You can change settings at runtime in the same way you do with `BaseModel`; in the constructor: `t2l = TextToLatentInvocation(steps=42)`. Is this what you're asking about?

I have written tests that check whether you can assign invalid fields to invocation objects, as well as checking that type coercion (e.g. str to Path) is working the way that pydantic does, and all seems OK. Graphs seem to be working.

The command line parsing code is what you wrote. I simply moved it from cli_app into the InvokeAISettings app so that each invocation object knows how to parse its own commands.